packages/rocketchat-api/server/v1/channels.js

```diff
@@ -422,16 +422,19 @@ RocketChat.API.v1.addRoute('channels.members', { authRequired: true }, {
 		const findResult = findChannelByIdOrName({ params: this.requestParams(), checkedArchived: false });
 
 		const { offset, count } = this.getPaginationItems();
-		const { sort } = this.parseJsonQuery();
+		const { sort, fields } = this.parseJsonQuery();
 
 		const members = RocketChat.models.Rooms.processQueryOptionsOnResult(Array.from(findResult.usernames), {
 			sort: sort ? sort : -1,
 			skip: offset,
 			limit: count
 		});
 
+		const ourFields = Object.assign({ _id: 1, username: 1, status: 1 }, fields, RocketChat.API.v1.defaultFieldsToExclude);
+		const users = RocketChat.models.Users.find({ username: { $in: members } }, { fields: ourFields }).fetch();
```

Comment:
How heavy is this find on large rooms?
What about use cache for that?
```
RocketChat.models.Subscriptions.find({rid: 'GENERAL'}).forEach(item => console.log(item._user))
```

Reply:
It's limited to 50 by default but users can request more and if they request a ton, then that load is on them.